src/renderers/dom/shared/Danger.js

```diff
@@ -166,7 +166,7 @@ var Danger = {
     );
     invariant(markup, 'dangerouslyReplaceNodeWithMarkup(...): Missing markup.');
     invariant(
-      oldChild.tagName.toLowerCase() !== 'html',
+      oldChild.nodeName !== 'HTML',
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeName indicates that `nodeName` should be `'html'` for XHTML docs but that doesn't appear to be true in practice (for several years, the linked article there is from 2009) so this is probably safe.